utilities/generate_cluster_configs_from_vagrant_hosts.py

```diff
@@ -10,26 +10,28 @@
 from libraries.utilities.generate_clusters_from_pool import generate_clusters_from_pool
 
 
-def generate_cluster_configs_from_vagrant(private_network, public_network):
+def generate_cluster_configs_from_vagrant(private_network, public_network, public_network_ethernet):
     """
     1. Gets the status for a running vagrant vm set.
     2. Uses the host name to look up the ip allocated to each vagrant vm instance
     3. Uses this IP list to build a pool.json file and generate the cluster configurations
     """
 
-    if private_network and public_network:
-        raise ProvisioningError("Invalid private_network and public_network flags")
+    if private_network and public_network and public_network_ethernet:
```

Comment:
Add a different type of verification. What were are looking for is that only one flag is set and the other two are not. You could use a count to make sure only one is set or use a series of asserts.